src/components/molecules/UserStatusManager/components/UserStatusPanel/UserStatusPanel.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useCallback, useMemo } from "react";
+import { CirclePicker, ColorResult } from "react-color";
+import { faEyeDropper, faTimes } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import { UserStatus } from "types/User";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { InputField } from "components/atoms/InputField";
+
+import "./UserStatusPanel.scss";
+
+export interface UserStatusPanelProps {
+  userStatus: UserStatus;
+  disabled?: boolean;
+  onDelete?: () => void;
+  onPickColor?: (color: string) => void;
+  onChangeInput?: (value: React.FormEvent<HTMLInputElement>) => void;
+}
+
+export const UserStatusPanel: React.FC<UserStatusPanelProps> = ({
+  userStatus,
+  disabled = false,
+  onDelete,
+  onPickColor,
+  onChangeInput,
+}) => {
+  const {
+    isShown: isColorPickerShown,
+    hide: hideColorPicker,
+    toggle: toggleColorPicker,
+  } = useShowHide();
+
+  const pickColor = useCallback(
+    (value: ColorResult) => {
+      onPickColor && onPickColor(value.hex);
+      hideColorPicker();
+    },
+    [hideColorPicker, onPickColor]
+  );
+
+  const statusIndicatorStyles = useMemo(
+    () => ({ backgroundColor: userStatus.color }),
+    [userStatus.color]
+  );
+
+  return (
+    <>
+      <div className="UserStatusPanel__status">
+        <InputField
+          value={userStatus.status}
+          onChange={onChangeInput}
+          disabled={disabled}
+        />
+        <div
+          style={statusIndicatorStyles}
+          className="UserStatusPanel__picker-icon"
+          onClick={toggleColorPicker}
+        >
+          <FontAwesomeIcon icon={faEyeDropper} />
+        </div>
+        {!disabled && (
+          <div className="UserStatusPanel__delete-icon" onClick={onDelete}>
+            <FontAwesomeIcon icon={faTimes} />
+          </div>
+        )}
+      </div>
+      {isColorPickerShown && !disabled && <CirclePicker onChange={pickColor} />}
```

Comment:
let's have a boolean const for `isColorPickerShown && !disabled` here as well

Reply:
and can we please split sibling blocks with empty rows?

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1532/commits/0c1f9f07a5696848014d78ad2604de3c1ed2d194